File: image-crud/src/main/java/org/judexmars/imagecrud/config/minio/MinioConfig.java
package org.judexmars.imagecrud.config.minio;

import io.minio.MinioClient;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@Slf4j
@EnableConfigurationProperties(MinioProperties.class)
public class MinioConfig {

    @Bean
    public MinioClient minioClient(MinioProperties minioProperties) {
        log.info("MINIO CLIENT: " + minioProperties.toString());
        return MinioClient.builder()
                .credentials(minioProperties.getAccessKey(), minioProperties.getSecretKey())
                .endpoint(minioProperties.getUrl())
                .build();
    }
}

File: image-processor/src/main/kotlin/org/judexmars/imageprocessor/config/minio/MinioConfig.kt
package org.judexmars.imageprocessor.config.minio

import io.minio.MinioClient
import lombok.extern.slf4j.Slf4j
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * Minio configuration.
 */
@Configuration
@Slf4j
@EnableConfigurationProperties(
    MinioProperties::class
)
class MinioConfig {
    /**
     * Minio client.
     *
     * @param minioProperties minio properties
     * @return minio client
     */
    @Bean
    fun minioClient(minioProperties: MinioProperties): MinioClient {
        return MinioClient.builder()
            .credentials(minioProperties.accessKey, minioProperties.secretKey)
            .endpoint(minioProperties.url)
            .build()
    }
}
